runtime/src/bank.rs

```diff
@@ -1973,6 +1984,20 @@ impl Bank {
         debug_do_not_add_builtins: bool,
         accounts_data_len: u64,
     ) -> Self {
+        let now = Instant::now();
+        let ancestors = Ancestors::from(&fields.ancestors);
+        // For backward compatibility, we can only serialize and deserialize
+        // Stakes<Delegation> in BankFieldsTo{Serialize,Deserialize}. But Bank
+        // caches Stakes<StakeAccount>. Below Stakes<StakeAccount> is obtained
+        // from Stakes<Delegation> by reading the full account state from
+        // accounts-db. Note that it is crucial that these accounts are loaded
+        // at the right slot and match precisely with serialized Delegations.
+        let stakes = Stakes::new(&fields.stakes, |pubkey| {
+            let (account, _slot) = bank_rc.accounts.load_with_fixed_root(&ancestors, pubkey)?;
+            Some(account)
+        })
+        .unwrap();
```

Comment:
This will panic if the stakes cache and accountsdb are inconsistent right? Seems like we should should rebuild the cache in this case, what do you think?

Reply:
Well it is a major logic bug if these are inconsistent, and can indicate bugs in the cache or accounts-db or a corrupted snapshot.
I think having this panic here will actually help to detect such inconsistencies earlier.
Should this actually happen, I think the best would be to hold on going down this path until we can debug this inconsistency.

Reply:
>  I think having this panic here will actually help to detect such inconsistencies earlier.

This is true, can you change the `unwrap` to an `expect` with a message describing that it "can indicate bugs in the cache or accounts-db or a corrupted snapshot"

>  Should this actually happen, I think the best would be to hold on going down this path until we can debug this inconsistency.

You mean try this on our nodes first to see if it's happening?

Reply:
> can you change the unwrap to an expect with a message describing that it "can indicate bugs in the cache or accounts-db or a corrupted snapshot"

done.

> >  Should this actually happen, I think the best would be to hold on going down this path until we can debug this inconsistency.
>
> You mean try this on our nodes first to see if it's happening?

For all nodes which submit metrics we can monitor panics through the panic hook:
https://github.com/solana-labs/solana/blob/6c5a3ca4a/metrics/src/metrics.rs#L442-L454

So any node which runs the master branch, ours or not ours, will do. And then testnet nodes once this is backported to testnet.

Reply:
Cool, sounds good